src/py/flwr/server/driver/driver_client_manager.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower DriverClientManager."""
+
+
+import threading
+import uuid
+from typing import Dict, List, Optional, Set
+
+from ..client_manager import ClientManager
+from ..client_proxy import ClientProxy
+from ..criterion import Criterion
+
+
+class DriverClientManager(ClientManager):
+    """Provides a pool of available clients."""
+
+    def __init__(self) -> None:
+        self._cv = threading.Condition()
+        self.cids_to_ids: Dict[str, int] = {}
+        self.ids_to_cps: Dict[int, ClientProxy] = {}
```

Comment:
How about using a single Dict here of type `Dict[str, Tuple[int, ClientProxy]]` to avoid anything getting out of sync?

Reply:
Sure, we can do that